script/tool/test/custom_test_command_test.dart

```diff
@@ -128,7 +143,8 @@ void main() {
       ]);
 
       processRunner.mockProcessesForExecutable['dart'] = <io.Process>[
-        MockProcess(exitCode: 1),
+        MockProcess(exitCode: 0), // pub get
+        MockProcess(exitCode: 1), // test script
```

Comment:
This is a fragile.  It would be nice if mockProcessesForExecutable was a `Map<String, List<Process>>`  Where the key is the name of the process and the value is a list to handle multiple invocations of the same process.  No need to fix it here if it is a lot of work, just wanted to call it out.

Reply:
> This is a fragile.

Indeed, it's been on my list to get back to at some point.

> It would be nice if mockProcessesForExecutable was a `Map<String, List<Process>>` Where the key is the name of the process and the value is a list to handle multiple invocations of the same process.

That's exactly what it is, actually (now; it used to just be a `List<Process>`, and this was the incremental improvement I made a while back because that was *very* fragile, and seriously impaired my ability to build out meaningful test coverage of the tool during that effort). The problem that's left is that it's using a list instead of an argument matcher, so we can't distinguish between `dart pub get` and `dart run tool/run_tests.dart`, and still have to handle that positionally instead.